### DISCUSSION TITLE
How to fetch data in a server side component?

### DISCUSSION THREAD

**Discussion by simpel** [vercel/next.js#51498] [App Router] [answered]
Hi all!

What is the correct way of fetching data in a server side component? Perhaps a obvious question but I cant get my head around it. 

Lets say I have a page.tsx in the app router that uses a component and within that component I want to make a request, like the one below. This will make the component async so I cant use it as a component (getting `Its return type 'Promise<Element>' is not a valid JSX element.`). 

Since I want to keep it a server component I cant really use useEffect or something like that either. So how to solve this? Do I need to pass the data from the page.tsx file all the time or is there some other way of doing this?

```typescript
const HomePage = async ({ page }: THomePage) => {
	const dictionary = await getDictionary()
        return (<p>{dictionary.welcome}</p>)
}
```

**Comment by a4lmamun** [comment]
You may try to export like `export default HomePage as unknown as () => JSX.Element;`. I have solved same issue once by this way, there might be other ways too.

**Reply by icyJoseph** [reply]
That will cause Fast Refresh to be un-optimal.

- https://nextjs.org/docs/architecture/fast-refresh#limitations

> Anonymous arrow functions like `export default () => <div />;` cause Fast Refresh to not preserve local component state.

**Comment by johnrahme** [comment]
Hey! 
I have found that you can use 'use'
`import { use } from "react";`
 `const dictionary = use(getDictionary())`
Not sure if it's the correct way of doing it though but works.

**Comment by simpel** [comment]
> Hey! I have found that you can use 'use' `import { use } from "react";` `const dictionary = use(getDictionary())` Not sure if it's the correct way of doing it though but works.

I've tried this too but I'm actually not sure if its correct. When reading the docs I have a feeling that use() makes it a client component... but really not sure. And also when using `use` my tsc checks gets sad:

```
✖ tsc-files --noEmit --pretty declarations.d.ts:
src/templates/HomePage/HomePage.tsx
Module '"react"' has no exported member 'use'.
```

**Comment by icyJoseph** [comment, marked as answer]
Hi, 

You just need to use the latest React types and TypeScript 5.1, which now supports the JSX as promise.

Otherwise, whenever you used your Server Component you'd put `// @ts-expect-error ServerComponent` over it, the ServerComponent string was just a convention, it can be whatever.

```tsx
<div>
  {/* @ts-expect-error some error for server component */}
  <ServerComponentThatIsAsync />
</div>
```

**Reply by icyJoseph** [reply]
@simpel link to docs for first solution: https://nextjs.org/docs/app/building-your-application/configuring/typescript#async-server-component-typescript-error

**Reply by simpel** [reply]
Oh, thanks @icyJoseph!
